crates/sui-core/src/epoch/randomness.rs

```diff
@@ -351,10 +441,60 @@ impl RandomnessManager {
         Ok(())
     }
 
+    fn complete_dkg(
+        &self,
+        dkg_version: u64,
+        used_processed_messages: &VersionedUsedProcessedMessages,
+    ) -> FastCryptoResult<Output<PkG, EncG>> {
+        let confirmations = self.confirmations.values();
+        let rng = &mut StdRng::from_rng(OsRng).expect("RNG construction should not fail");
+
+        match dkg_version {
```

Comment:
again wondering can this be a method on VersionedUsedProcessedMessages? just the part that farms out the call to `complete` or `complete_v1` based on the input message type